extensions/wikia/AdEngine/js/slot/bottomLeaderboard.js

```diff
@@ -26,11 +39,35 @@ define('ext.wikia.adEngine.slot.bottomLeaderboard', [
 			if (!pushed && pushPos < scrollPosition) {
 				pushed = true;
 				doc.removeEventListener('scroll', pushSlot);
-				win.adslots2.push(slotName);
-				log(['pushSlot', 'Pushed ' + slotName], 'debug', logGroup);
+
+				if (!collapseOnSameViewport()) {
+					win.adslots2.push(slotName);
+					log(['pushSlot', 'Pushed ' + slotName], 'debug', logGroup);
+				}
 			}
 		});
 
+	function collapseOnSameViewport() {
+		if (adContext.get('opts.isBLBViewportEnabled') && isUap() && viewportObserver.sameViewport(slotName, breakSlots)) {
+			eventDispatcher.dispatch('adengine.slot.status', {
+				slot: {
```

Comment:
I believe it will hurt us that we're dispatching this or similar events with objects that have different interfaces